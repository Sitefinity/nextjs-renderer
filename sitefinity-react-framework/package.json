{
  "name": "sitefinity-react-framework",
  "version": "0.1.0",
  "private": true,
  "publishConfig": {
    "access": "public"
  },
  "license": "See license in EULA.md",
  "dependencies": {
    "@types/react": "18.2.28",
    "react": "18.2.0",
    "tslib": "2.3.1",
    "typescript": "5.2.2"
  },
  "scripts": {
    "build": "tsc"
  }
}
