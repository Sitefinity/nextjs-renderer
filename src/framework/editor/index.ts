export * from './renderer-contract-interfaces';
export * from './services';
export * from './utils';
export * from './widgets';

