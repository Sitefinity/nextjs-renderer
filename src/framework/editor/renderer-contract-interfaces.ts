import { WidgetModel } from "../widgets/widget-model";

export interface RendererContract {
    getWidgetMetadata(args: GetWidgetMetadataArgs): Promise<ComponentMetadata>;
    renderWidget(args: RenderWidgetArgs): Promise<RenderResult>;
    getCategories(args: GetCategoriesArgs): Promise<Array<string>>;
    getWidgets(args: GetWidgetsArgs): Promise<TotalCountResult<Array<WidgetSection>>>;
}

export interface TotalCountResult<T> {
    dataItems: T;
    totalCount: number;
}

export interface RenderResult {
    element: HTMLElement,
    content: string,
    scripts: Array<Script>
}

export interface Script {
    src: string;
    id: string;
}

export interface WidgetItem {
    name: string;
    title?: string;
    addWidgetTitle?: string;
    addWidgetName?: string;
    thumbnailUrl?: string;
    initialProperties?: Array<{ name: string, value: string}>
}

export interface WidgetSection {
    title: string;
    widgets: WidgetItem[];
}

export interface RenderWidgetArgs {
    dataItem: DataItem;
    siteId: string;
    model: WidgetModel<any>;
    token?: Token;
}
export interface GetCategoriesArgs {
    token?: Token;
    toolbox?: string;
}

export interface GetWidgetsArgs {
    dataItem: DataItem;
    category: string;
    search: string;
    skip: number;
    take: number;
    token?: Token;
    toolbox?: string;
}

export interface GetWidgetMetadataArgs {
    dataItem: DataItem;
    widgetKey: string;
    widgetName: string;
    siteId: string;
    token?: Token;
}

export interface ComponentMetadata {
    Caption: string,
    Name: string,
    PropertyMetadata: SectionGroup[],
    PropertyMetadataFlat: PropertyMetadata[],
}

export interface PropertyMetadata {
    Name: string;
    Title: string;
    Type: string;
    DefaultValue: string;
    Properties: { [key: string]: any }
}

export interface SectionGroup {
    Name: string;
    Sections: SectionData[];
}

export interface SectionData {
    Name: string,
    Title: string
    Properties: Array<PropertyMetadata>
}

export interface Token {
    type: string,
    value: string
}

export interface DataItem {
    readonly provider: string;
    readonly culture: string;
    readonly key: string;
}
