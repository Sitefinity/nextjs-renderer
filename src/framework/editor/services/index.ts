export * from './detail-item';
export * from './render-widget-service';
export * from './request-context';
