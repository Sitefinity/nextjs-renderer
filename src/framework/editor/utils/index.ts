export * from './classNames';
export * from './getUniqueId';
export * from './guid';
export * from './rest-extensions';
