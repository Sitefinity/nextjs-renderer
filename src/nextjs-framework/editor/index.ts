export * from './renderer-contract-interfaces';
export * from './utils/classNames';
export * from './utils/getUniqueId';
export * from './utils/guid';

export * from './detail-item';
export * from './request-context';

export * from './widget-framework/attributes';
export * from './widget-framework/content-list-settings';
export * from './widget-framework/link-model';
export * from './widget-framework/mixed-content-context';
export * from './widget-framework/widget-context';
export * from './widget-framework/widget-editor-metadata';
export * from './widget-framework/widget-metadata';
export * from './widget-framework/widget-model';
export * from './widget-framework/widget-registry';
