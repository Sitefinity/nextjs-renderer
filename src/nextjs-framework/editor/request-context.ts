import { PageLayoutServiceResponse, SdkItem } from '../rest-sdk';
import { DetailItem } from './detail-item';

export interface RequestContext {
    layout: PageLayoutServiceResponse;
    searchParams: { [key: string]: string; };
    detailItem?: DetailItem;
    culture: string;
    isEdit: boolean;
    isPreview: boolean;
    isLive: boolean;
}
