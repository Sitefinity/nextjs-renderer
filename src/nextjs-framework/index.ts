export * from './pages/render-page';
export * from './pages/render-widget';
export * from './pages/utils';
