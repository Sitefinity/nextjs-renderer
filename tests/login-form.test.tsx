import { LoginForm, LoginFormEntity } from '@progress/sitefinity-react-framework';
import { render, waitFor } from '@testing-library/react';
import { RequestContext, WidgetModel } from '../src/nextjs-framework/editor';

it('should render login form', async () => {
    const contentList = await LoginForm({ model, requestContext, metadata });
    const { container } = render(contentList);

    const labels = container.getElementsByTagName('label');
    for (let label of labels) {
        const value = label.getAttribute('for');
        if (value !== null) {
            label.setAttribute('for', `${value.split('--')[0]}--f754`);
        }
    }

    const inputs = container.getElementsByTagName('input');
    for (let input of inputs) {
        const value = input.getAttribute('id');
        if (value !== null) {
            input.setAttribute('id', `${value.split('--')[0]}--f754`);
        }
    }

    await waitFor(() => {
        expect(container).toMatchSnapshot();
    });
});

const metadata = {
    designerMetadata: {
        Name: 'SitefinityLoginForm',
        Caption: 'Login form',
        PropertyMetadata: [Array]
    },
    componentType: [LoginForm],
    editorMetadata: { Title: 'LoginForm' },
    ssr: true
};

const model: WidgetModel<LoginFormEntity> = {
    Id: 'b116390e-23ae-410a-b323-6354a04aaddc',
    Name: 'SitefinityLoginForm',
    PlaceHolder: 'Column1',
    Caption: '',
    ViewName: '',
    Lazy: false,
    Properties: { Margins: undefined, RememberMe: true },
    Children: []
};

const requestContext: RequestContext = {
    layout: {
        '@odata.context': 'https://localhost:5001/api/default/$metadata#Telerik.Sitefinity.Renderer.Web.Services.Dto.PageDtoWithContext',
        Culture: 'en',
        SiteId: '4c922118-f076-4e24-9193-93e004f50107',
        MetadataHash: '-1129976431',
        Id: '831706cd-10e1-4184-b2e5-ef9f9ad0fdb5',
        TemplateName: 'Default',
        Renderer: 'React',
        UrlParameters: [],
        HasVariations: false,
        Site: {
            Id: '4c922118-f076-4e24-9193-93e004f50107',
            Name: 'Quantum International',
            DefaultCulture: 'en',
            Cultures: [Array],
            TimeZone: 'UTC;0;UTC;UTC;UTC;;',
            IsSubFolder: false
        },
        User: null,
        ComponentContext: {
            HasLazyComponents: false,
            OrphanedControls: [],
            Components: [Array],
            Fields: null,
            Site: null,
            User: null
        },
        Scripts: [],
        DetailItem: null,
        MetaInfo: {
            Title: 'integration-test',
            Description: '',
            HtmlInHeadTag: null,
            OpenGraphTitle: 'integration-test',
            OpenGraphDescription: '',
            OpenGraphImage: null,
            OpenGraphVideo: null,
            OpenGraphType: null,
            OpenGraphSite: null,
            CanonicalUrl: 'https://localhost:5001/integration-test'
        },
        Fields: {
            Title: 'integration-test',
            UrlName: 'integration-test',
            LastModified: '2023-11-13T13:13:09.183Z',
            DateCreated: '2023-11-13T13:13:09.183Z',
            Crawlable: true,
            ParentId: 'f669d9a7-009d-4d83-ddaa-000000000002',
            RootId: 'f669d9a7-009d-4d83-ddaa-000000000002',
            IsHomePage: false,
            'AvailableLanguages@odata.type': '#Collection(String)',
            AvailableLanguages: [Array],
            ViewUrl: '/integration-test'
        }
    },
    searchParams: {
        sf_site: '4c922118-f076-4e24-9193-93e004f50107',
        sfaction: 'edit',
        sf_version: '80'
    },
    detailItem: null,
    culture: 'en',
    isEdit: true,
    isPreview: false,
    isLive: false,
    cookie: undefined
};
