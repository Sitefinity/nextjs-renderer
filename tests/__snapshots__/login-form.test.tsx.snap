// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login form 1`] = `
<div>
  <div
    class=""
    data-sfemptyiconaction="Edit"
    data-sfhasquickeditoperation="true"
    data-sfid="b116390e-23ae-410a-b323-6354a04aaddc"
    data-sfiscontentwidget="true"
    data-sfisempty="false"
    data-sfisemptyvisualhidden="false"
    data-sfisorphaned="false"
    data-sfname="SitefinityLoginForm"
    data-sftitle="LoginForm"
    draggable="true"
  >
    <div
      data-sf-role="form-container"
    >
      <h2
        class="mb-3"
      >
        Login
      </h2>
      <div
        aria-live="assertive"
        class="alert alert-danger my-3 d-none"
        data-sf-role="error-message-container"
        id="errorContainer"
        role="alert"
        style="display: none;"
      >
        Incorrect credentials.
      </div>
      <form
        action="/sitefinity/login-handler"
        method="post"
        novalidate=""
        role="form"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="sf-username--f754"
          >
            Email / Username
          </label>
          <input
            class="form-control"
            data-sf-role="required"
            id="sf-username--f754"
            name="username"
            type="email"
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="sf-password--f754"
          >
            Password
          </label>
          <input
            autocomplete="on"
            class="form-control"
            data-sf-role="required"
            id="sf-password--f754"
            name="password"
            type="password"
          />
        </div>
        <div
          class="row mb-3"
        >
          <div
            class="checkbox col-md-6 m-0"
          >
            <label>
              <input
                checked=""
                data-val="true"
                data-val-required="The RememberMe field is required."
                id="sf-rememeber--f754"
                name="RememberMe"
                type="checkbox"
                value="true"
              />
              <label
                for="sf-rememeber--f754"
              >
                Remember me
              </label>
            </label>
          </div>
        </div>
        <input
          name="RedirectUrl"
          type="hidden"
          value="https://localhost:5001/integration-test?sf_site=4c922118-f076-4e24-9193-93e004f50107&sfaction=edit&sf_version=80"
        />
        <input
          name="ErrorRedirectUrl"
          type="hidden"
          value="https://localhost:5001/integration-test?sf_site=4c922118-f076-4e24-9193-93e004f50107&sfaction=edit&sf_version=80&loginerror=True"
        />
        <input
          name="MembershipProviderName"
          type="hidden"
          value=""
        />
        <input
          name="sf_antiforgery"
          type="hidden"
          value=""
        />
        <input
          class="btn btn-primary w-100"
          type="submit"
          value="Log in"
        />
      </form>
      <input
        name="ValidationInvalidEmailMessage"
        type="hidden"
        value="Invalid email format."
      />
      <input
        name="ValidationRequiredMessage"
        type="hidden"
        value="All fields are required."
      />
    </div>
  </div>
</div>
`;
